### PR TITLE
Implement `clustfcc` module

### PR DESCRIPTION
**Pull request by rvhonorato**
This PR implements the last pending core module after `beta1` migration (#32) by adding the `clustfcc` module.

For it to work in the workflow philosophy I also had to add a `seletopclust` "helper module" that handles the selection;

the `seletopclusts` takes two arguments, the ranking of the clusters to be selected and how many models of each. It uses the `pdb.score` taken from the `previous_io` to calculate the cluster and model rankings.

```toml
[seletopclust]
top_cluster = [1, 2, 3, 4] # will select the top4 clusters
top_models = 4 # will select the best 4 models inside each cluster
```

This effectively means we can now do things like:

`topology > rigid-body > fcc-clustering > select-top-cluster > flexible-refinement > md-refinement`

***

I also added `haddocking/fcc` as a submodule, pulling from the python3 branch to avoid code repetition.

### CODE REVIEW

**Review by joaomcteixeira — CHANGES_REQUESTED** [review]
Apart from my little comments, if it works, it looks good. :+1:

File: src/haddock/engine.py [hunk 1]
```diff
@@ -1,10 +1,38 @@
 """Running CNS scripts"""
 import subprocess
-from haddock.error import CNSRunningError
+from haddock.error import CNSRunningError, JobRunningError
```
Comment (joaomcteixeira) [inline src/haddock/engine.py:3]: In #66 this file will be in `libs`, no problem. :+1:

File: src/haddock/engine.py [hunk 2]
```diff
@@ -1,10 +1,38 @@
...
+
+        # capture output to stdout
+        cmd += f' {self.input}'
+        with open(self.output, 'w') as outf:
+            p = subprocess.Popen(shlex.split(cmd), 
+                                 stdout=outf,
+                                 close_fds=True)
+            out, error = p.communicate()
+        p.kill()
+        outf.close()
```
Comment (joaomcteixeira) [inline src/haddock/engine.py:30]: you don't need `outf.close()` is already with `with`.

File: src/haddock/engine.py [hunk 3]
```diff
@@ -1,10 +1,38 @@
...
+        self.executable = executable
+        self.args = args
+
+    def run(self):
+        if self.args:
+            cmd = f"{self.executable} {' '.join(map(str, self.args))}"
+            self.executable = shlex.split(cmd)
+
+        # capture output to stdout
+        cmd += f' {self.input}'
```
Comment (joaomcteixeira) [inline src/haddock/engine.py:23]: I think here there's a bug if `args` are not given because `cmd` won't be defined.

Comment (rvhonorato) [reply src/haddock/engine.py:23]: oh yeh, indeed!

**Review by joaomcteixeira — APPROVED** [review]
Look good :relaxed:

**Pull request merged**